### PR TITLE
GraphQL Data Connector E2E tests

### PR DESCRIPTION
**Pull request by gloomweaver**

### CODE REVIEW

File: .github/workflows/e2e_test_ci.yml [hunk 1]
```diff
@@ -272,6 +272,89 @@ jobs:
...
+        - name: Install spice
+          uses: ./.github/actions/install-spice
+          with:
+            build-path: ./build
+
+        - name: Init spice app
+          run: |
+            spice init test_app
+
+        - name: Spice dataset configure
```
Comment (sgrebnov) [inline .github/workflows/e2e_test_ci.yml:306]: @gloomweaver Kirill please verify this step:
https://github.com/spiceai/spiceai/actions/runs/9323632524/job/25667579082

>Run echo -e "test_graphql_table\nrepos\graphql:[https://api.github.com/graphql\ny](https://api.github.com/graphql/ny)" | spice dataset configure
dataset name: (test_app) description: from: locally accelerate (y/n)? (y) EOF
Error: Process completed with exit code 1.

Comment (sgrebnov) [reply .github/workflows/e2e_test_ci.yml:306]: I think it is due to missing 'n' here: `\nrepos\graphql `
Should be
>echo -e "test_graphql_table\nrepos\ngraphql:https://api.github.com/graphql\ny" | spice dataset configure

File: .github/workflows/e2e_test_ci.yml [hunk 2]
```diff
@@ -272,6 +272,72 @@ jobs:
...
+          working-directory: test_app
+          run: |
+            echo -e "test_graphql_table\nrepos\ngraphql:https://api.github.com/graphql\ny" | spice dataset configure
+            # configure graphql credentials
+            echo -e "params:\n  auth_token_key: token\n  json_path: data.viewer.repositories.nodes\n  query: \"{ viewer { repositories(first: 100) { nodes { id name watchers (first: 10) { nodes { name } } } } } }\"" >> ./datasets/test_graphql_table/dataset.yaml
+            # configure env secret store
+            echo -e "secrets:\n  store: env\n" >> spicepod.yaml
+            cat spicepod.yaml
```
Comment (ewgenius) [inline .github/workflows/e2e_test_ci.yml:314]: E2E CI tests shouldn't depend on external services, that might block PR's. Let's replace that with local graphql app in docker.

Comment (gloomweaver) [reply .github/workflows/e2e_test_ci.yml:314]: It runs only on workflow_dispatch event

Comment (ewgenius) [reply .github/workflows/e2e_test_ci.yml:314]: what is the point of that test then?

Comment (ewgenius) [reply .github/workflows/e2e_test_ci.yml:314]: `workflow_dispatch` covers only cases where we can't make local setup (dremio and spiceai from quickstarts)

Comment (ewgenius) [reply .github/workflows/e2e_test_ci.yml:314]: adding simple node.js app with graphql endpoint and some static data response should be pretty straightforward

Comment (gloomweaver) [reply .github/workflows/e2e_test_ci.yml:314]: Yeah, I agree

**Comment by gloomweaver** [comment]
Will setup integration testing against separate graphql server in a separate pr.

**Pull request closed without merging**